### PR TITLE
[fides.js] Fix button arrangement

### PR DESCRIPTION
**Pull request by allisonking**
Closes https://ethyca.atlassian.net/browse/PROD-1339

### Description Of Changes

Privacy policy is now always in the middle, per this design: https://www.figma.com/file/dmEwdK3xZwjKfGVQ9t66xe/Fides-v.2-Master-Working-Files?type=design&node-id=11310-269887&mode=design&t=OnhFjNC97D3WXHL9-0

This simplifies the code considerably! Also fixes some alignment issues.

https://www.loom.com/share/d27a84f9a8a241c4bbe43e22a13c5500?sid=bee861e1-df2c-4747-9427-ce5da6022d2f

### Code Changes

* [x] Have `ConsentBanner.tsx` be responsible for rendering the privacy policy instead of parent components passing it in as a middle button or as a child. Since the policy is always in the middle now, this is feasible and cleans the code up a bit!

### Steps to Confirm

* [ ] See loom—test out having a privacy policy and not having a privacy policy, at different viewport sizes, and for both TCF and notices

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* [x] Issue Requirements are Met
* [x] Update `CHANGELOG.md`

### CODE REVIEW

File: clients/fides-js/src/components/fides.css [hunk 1]
```diff
@@ -67,6 +67,7 @@
   --fides-overlay-container-border-radius: 12px;
   --fides-overlay-component-border-radius: 4px;
   --fides-overlay-banner-offset: 48px;
+  --fides-banner-font-size-title: var(--16px);
```
Comment (allisonking) [inline clients/fides-js/src/components/fides.css:70]: I removed this variable earlier not realizing the nyt extension was using it, so this is restoring it!

**Comment by Kelsey-Ethyca** [comment]
- TCF Enabled
   - banner
      - privacy policy set
         - mobile
            there is not a design for this but I have asked Simon to add one
         - full width ❌
            in the design the order is manage policy accept reject 
            the order on this branch is manage policy reject accept
      - privacy policy link not set 
         - mobile
            there is not a design for this but I have asked Simon to add one
         - full width ❌
            in the design the order is manage policy accept reject 
            the order on this branch is manage policy reject accept
     - modal
       - privacy policy set
         - mobile
            there is not a design for this but I have asked Simon to add one
         - full width ✅ 
      - privacy policy link not set 
         - mobile
            there is not a design for this but I have asked Simon to add one
         - full width ✅ 

- TCF Disabled
   - banner
      - privacy policy set
         - mobile
            there is not a design for this but I have asked Simon to add one
         - full width ✅ 
      - privacy policy link not set 
         - mobile
            there is not a design for this but I have asked Simon to add one
         - full width ✅ 
     - modal
       - privacy policy set
         - mobile
            there is not a design for this but I have asked Simon to add one
         - full width ✅ 
      - privacy policy link not set 
         - mobile
            there is not a design for this but I have asked Simon to add one
         - full width  ✅

**Comment by galvana** [comment]
Code-wise this looks good, but I noticed some spacing issues

The buttons have some left padding that causes the Privacy Policy to be a little off center
image

The TCF modal seems to have more padding on the right side than the left
image

File: clients/fides-js/src/components/ConsentButtons.tsx [hunk 2]
```diff
@@ -70,8 +68,6 @@ interface NoticeConsentButtonProps {
   enabledKeys: NoticeKeys;
   isAcknowledge: boolean;
   isInModal?: boolean;
-  children?: ComponentChildren;
-  middleButton?: VNode;
```
Comment (galvana) [inline clients/fides-js/src/components/ConsentButtons.tsx:74]: 🙌 I'm glad these are gone!

**Comment by allisonking** [comment]
Thanks @galvana , that's a good catch! I'll fix the opt in/out button padding 👍 

I'm unable to reproduce the modal padding being different though... I wonder if it has to do with that scrollbar thing? 
image

**Comment by allisonking** [comment]
Ah yeah I just turned my "Show scroll bars" in osx settings to "Always" and now I see it...
image

**Comment by allisonking** [comment]
@galvana it appears that removing `scrollbar-gutter: stable` from `.fides-modal-body` fixes it for me, but I know you added that for a reason... but I forget what it was!

**Comment by allisonking** [comment]
Fixed the button padding issue! [c0b6e0b](https://github.com/ethyca/fides/pull/4407/commits/c0b6e0b31ab260bf0d16049d8223458b47dfb0b9)
image

**Comment by galvana** [comment]
> @galvana it appears that removing `scrollbar-gutter: stable` from `.fides-modal-body` fixes it for me, but I know you added that for a reason... but I forget what it was!

@allisonking, it was so that the modal description didn't flicker when going from a tab with overflow content to one without. I think we can just move `scrollbar-gutter: stable` from `fides-modal-body` to the new outermost container, `fides-consent-content`. It worked for me locally

scroll

**Review by galvana — APPROVED** [review]
Thanks for the fixes, this looks ready to me!

**Pull request merged**